src/Search/Search.js

```diff
@@ -14,43 +17,109 @@ function WikipediaSearch() {
+    const url = 'https://en.wikipedia.org/w/api.php?'
+    let params = {
+      action: 'query',
+      list: 'search',
+      format: 'json',
+      origin: '*',
+      srlimit: 20,
+      srsearch: searchTerm,
+    };
+
+    await fetch(url + new URLSearchParams(params).toString())
       .then((response) => {
         if (!response.ok) {
           throw new Error('Network response was not ok');
         }
         return response.json();
       })
       .then((data) => {
-        const filteredResults = data.query.search.filter(
-          (item) =>
-            item.title.toLowerCase().includes('abuse') ||
-            item.snippet.toLowerCase().includes('abuse')||
-            item.snippet.toLowerCase().includes('allegation') ||
-            item.snippet.toLowerCase().includes('drug') ||
-            !containsHtmlTags(item.snippet.toLowerCase()) // Check if snippet contains HTML tags
-        );
-        // Clean up the snippet text by removing HTML tags and their content
-        filteredResults.forEach((result) => {
-          result.snippet = result.snippet.replace(/<[^>]*>?/gm, '');
-        });
-        setResults(filteredResults);
-        console.log(filteredResults)
+// Sets the state of the initial results to the first result of the search (i.e. Rush Limbaugh)
+        setInitialResults(data.query.search[0])
       })
       .catch((error) => {
         console.error('There was a problem with the fetch operation:', error);
       });
   };
 
-  const containsHtmlTags = (str) => {
-    const htmlTagsRegex = /<([A-Za-z][A-Za-z0-9]*)\b[^>]*>(.*?)<\/\1>|<([A-Za-z][A-Za-z0-9]*)\b[^\/]*\/>/;
-    return htmlTagsRegex.test(str);
-  };
+// This useEffect will run when the initialResults state is updated to find the page from the initial result  
+  useEffect(() => {
+    
+      const url = 'https://en.wikipedia.org/w/api.php?'
+      let params = {
+        action: 'parse',
+        prop: 'sections',
+        format: 'json',
+        origin: '*',
+        pageid: initialResults.pageid,
+      };
+      fetch(url + new URLSearchParams(params).toString())
+        .then((response) => {
+          if (!response.ok) {
+            throw new Error('Network response was not ok');
+          }
+          return response.json();
+        })
+        .then((data) => {
+          setPage(data.parse.sections)
+        })
+        .catch((error) => {
+          console.error('There was a problem with the fetch operation:', error);
+        });
+    
+  }, [initialResults]);
+
+// This useEffect will run when the page state is updated to find the controversies
+  useEffect(() => {
+    const words = ["Controversies", "Controversy", "Hoax", "Criticism", "Scandal", "Legal Issues", "Conspiracy"];
+
+// This function will find the sections that contain the words in the words array
+    function findMatchingSections(page, words) {
```

Comment:
loved that you could take the set up I had and improve it so much in a way I didn't even consider. you really had some good solutions to making the functions way more dry.